modules/plugin-search.php

```diff
@@ -0,0 +1,502 @@
+			add_filter( 'self_admin_url', array( $this, 'plugin_details' ) );
+			add_filter( 'plugin_install_action_links', array( $this, 'insert_module_related_links' ), 10, 2 );
+		}
+	}
+
+	/**
+	 * Modify URL used to fetch to plugin information so it pulls Jetpack plugin page.
+	 *
+	 * @param string $url URL to load in dialog pulling the plugin page from wporg.
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return string The URL with 'jetpack' instead of 'jetpack-plugin-search'.
+	 */
+	public function plugin_details( $url ) {
+		return false !== stripos( $url, 'tab=plugin-information&amp;plugin=' . self::$slug )
+			? 'plugin-install.php?tab=plugin-information&amp;plugin=jetpack&amp;TB_iframe=true&amp;width=600&amp;height=550'
+			: $url;
+	}
+
+	/**
+	 * Register REST API endpoints.
+	 *
+	 * @since 7.1.0
+	 */
+	public static function register_endpoints() {
+		register_rest_route( 'jetpack/v4', '/hints', array(
+			'methods' => WP_REST_Server::EDITABLE,
+			'callback' => __CLASS__ . '::dismiss',
+			'permission_callback' => __CLASS__ . '::can_request',
+			'args' => array(
+				'hint' => array(
+					'default'           => '',
+					'type'              => 'string',
+					'required'          => true,
+					'validate_callback' => __CLASS__ . '::is_hint_id',
+				),
+			)
+		) );
+	}
+
+	/**
+	 * A WordPress REST API permission callback method that accepts a request object and
+	 * decides if the current user has enough privileges to act.
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return bool does a current user have enough privileges.
+	 */
+	public static function can_request() {
+		return current_user_can( 'jetpack_admin_page' );
+	}
+
+	/**
+	 * Validates that the ID of the hint to dismiss is a string.
+	 *
+	 * @since 7.1.0
+	 *
+	 * @param string|bool $value Value to check.
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 * @param string $param Name of the parameter passed to endpoint holding $value.
+	 *
+	 * @return bool|WP_Error
+	 */
+	public static function is_hint_id( $value, $request, $param ) {
+		return in_array( $value, Jetpack::get_available_modules(), true )
+			? true
+			: new WP_Error( 'invalid_param', sprintf( esc_html__( '%s must be an alphanumeric string.', 'jetpack' ), $param ) );
+	}
+
+	/**
+	 * A WordPress REST API callback method that accepts a request object and decides what to do with it.
+	 *
+	 * @param WP_REST_Request $request {
+	 *     Array of parameters received by request.
+	 *
+	 *     @type string $hint Slug of card to dismiss.
+	 * }
+	 *
+	 * @since 7.1.0
+	 *
+	 * @return bool|array|WP_Error a resulting value or object, or an error.
+	 */
+	public static function dismiss( WP_REST_Request $request ) {
+		return self::add_to_dismissed_hints( $request['hint'] )
+			? rest_ensure_response( array( 'code' => 'success' ) )
+			: new WP_Error( 'not_dismissed', esc_html__( 'The card could not be dismissed', 'jetpack' ), array( 'status' => 400 ) );
```

Comment:
Translators were asking what type of card this is. Please add a context or a translator comment to clarify.